code/modules/atmospherics/machinery/pipes/pipes.dm

```diff
@@ -27,12 +27,9 @@
 	volume = 35 * device_type
 	. = ..()
 
-///I have no idea why there's a new and at this point I'm too afraid to ask
-/obj/machinery/atmospherics/pipe/Initialize(mapload)
-	. = ..()
-
-	if(hide)
-		AddElement(/datum/element/undertile, TRAIT_T_RAY_VISIBLE) //if changing this, change the subtypes RemoveElements too, because thats how bespoke works
+/obj/machinery/atmospherics/pipe/setup_hiding()
+	AddElement(/datum/element/undertile, TRAIT_T_RAY_VISIBLE) //if changing this, change the subtypes RemoveElements too, because thats how bespoke works
+	return ..() // Call the parent after, so undertile changes are before pipe cap updates
```

Comment:
I don't like this. order of operations should not be a concern we have with signals. (People should not need to think about this) 
Is there anything we can do about needing one to run before the other? do we want pipecaps to be based off the result of undertile?

Reply:
Honestly that's a good point, though the first issue I can think about with that is that not every `/obj/machinery/atmospherics` subtype *has* undertile, only plain pipes iirc, while a nonzero amount of them do have the pipe caps updating...

Maybe we could just, have it additionally register pipe cap updating to run after we add or remove the `TRAIT_UNDERFLOOR` trait? Given that's the primary thing undertile changes that it actually cares about.
So registering it to run on that would probably be the easiest solution.

Reply:
That said, it would still call it twice, so like, hmn...
Maybe just, have a proc just for registering the signal on which to update pipe caps? Where pipes override it to register it on a different signal, whether that's the underfloor trait check or a new one sent at the end of undertile.
I'll think about it some more later.